EarlyBird.BusinessLogic/Services/OffersService.cs

```diff
@@ -48,21 +68,48 @@ public ViewOfferDto Add(AddOfferDto offerEntity)
         public bool Delete(int id)
         {
             var offer = offersRepository.GetById(id);
-            if (offer == null) throw new EntityNotFoundException() ;
+            if (offer == null) throw new OfferNotFoundException("Offer could not be found");
 
             return offersRepository.Delete(offer);
         }
 
-        public bool Update(int id, UpdateOfferDto offerEntity)
+        public bool Update(int id, UpdateOfferDto updateOfferEntity)
+        {
+            var offerEntity = offersRepository.GetById(id);
+            if (offerEntity == null)
+                throw new OfferNotFoundException("Offer could not be found");
+
+            offerEntity.Title = updateOfferEntity.Title ?? offerEntity.Title;
+            offerEntity.Cost = updateOfferEntity.Cost == 0 ? offerEntity.Cost : updateOfferEntity.Cost;
+            offerEntity.PublisherId = updateOfferEntity.PublisherId;
```

Comment:
You should not be able to update the publisher

Reply:
nu se face update, ci se transmite id-ul pentru a fi salvat, in controller lui updateOfferEntity i se atribuie userul curent pe campul de publisher, daca puneam fara atribbuirea asta ramanea campul null si operatorii aia de null comperator nu mergeau pe Guid